src/Widgets/EventRow.vala

```diff
@@ -70,6 +70,15 @@ public class DateTime.EventRow : Gtk.ListBoxRow {
         grid_context.add_class ("event");
         grid_context.add_provider (css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
 
+        /* Color menuitem per calendar source of event */
+        Util.get_style_calendar_color (cal_event.cal, grid);
+        Util.get_style_calendar_color (cal_event.cal, event_image);
+
+        cal_event.cal.notify["color"].connect (() => {
+            Util.get_style_calendar_color (cal_event.cal, grid);
+            Util.get_style_calendar_color (cal_event.cal, event_image);
```

Comment:
DRY this please

Reply:
How do I do it, it should run the method for both, otherwise the icon never gets the color.